src/pages/DetailsPage/__tests__/DetailsPage.test.tsx

```diff
@@ -102,64 +103,54 @@ describe('Details Page', () => {
     const isLoading = true;
     const isOrgAdmin = true;
     mockAuthenticateUser(isLoading, isOrgAdmin, true);
-
-    const { container } = render(<Page />);
-
-    await waitFor(() => expect(useUser).toHaveBeenCalledTimes(1));
-    expect(container).toMatchSnapshot();
-  });
-
-  it('renders data', async () => {
-    const isLoading = false;
-    const isOrgAdmin = true;
-    const canReadProducts = true;
-    mockAuthenticateUser(isLoading, isOrgAdmin, canReadProducts);
-    mockSingleProduct(true);
-
-    const { container } = render(<Page />);
-
-    await waitFor(() => expect(useSingleProduct).toHaveBeenCalledTimes(1));
-    expect(container).toMatchSnapshot();
+    const container = render(<Page />);
+    expect(container).toHaveLoader();
   });
+});
 
-  it("redirects when can't read products", async () => {
-    const isLoading = false;
-    const isOrgAdmin = true;
-    const canReadProducts = false;
-    mockAuthenticateUser(isLoading, isOrgAdmin, canReadProducts);
-    mockSingleProduct(false);
-
-    const { container } = render(<Page />);
+it('renders data', async () => {
+  const isLoading = false;
+  const isOrgAdmin = true;
+  const canReadProducts = true;
+  mockAuthenticateUser(isLoading, isOrgAdmin, canReadProducts);
+  mockSingleProduct(true);
 
-    await waitFor(() => expect(useSingleProduct).toHaveBeenCalledTimes(2));
-    expect(container).toMatchSnapshot();
+  const { getAllByText } = render(<Page />);
+  getAllByText('1234').forEach((el) => {
+    expect(el).toBeInTheDocument();
   });
+});
 
-  it('renders not available for missing data', async () => {
-    const isLoading = false;
-    const isOrgAdmin = true;
-    const canReadProducts = true;
-    mockAuthenticateUser(isLoading, isOrgAdmin, canReadProducts);
-    mockSingleProduct(false);
-
-    const { container } = render(<Page />);
+it("redirects when can't read products", async () => {
+  const isLoading = false;
+  const isOrgAdmin = true;
+  const canReadProducts = false;
+  mockAuthenticateUser(isLoading, isOrgAdmin, canReadProducts);
+  mockSingleProduct(false);
 
-    await waitFor(() => expect(useSingleProduct).toHaveBeenCalledTimes(1));
-    expect(container).toMatchSnapshot();
-  });
+  render(<Page />);
+  expect(useSingleProduct).toHaveBeenCalledTimes(2);
+});
 
-  it('handles errors', async () => {
-    (useSingleProduct as jest.Mock).mockReturnValue({
-      isLoading: false,
-      isFetching: false,
-      isSuccess: false,
-      error: true,
-      data: []
-    });
+it('renders not available for missing data', async () => {
+  const isLoading = false;
+  const isOrgAdmin = true;
+  const canReadProducts = true;
+  mockAuthenticateUser(isLoading, isOrgAdmin, canReadProducts);
+  mockSingleProduct(false);
 
-    const { container } = render(<Page />);
+  render(<Page />);
+  expect(document.querySelector('.pf-c-breadcrumb').firstChild.textContent);
```

Comment:
This doesn't have any expectation - you need to check that is matches not available